cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetUpgrader.java

```diff
@@ -16,12 +16,22 @@
 
 package co.cask.cdap.data.tools;
```

Comment:
Shouldn't this class/logic be in data-fabric (i.e. in datasets module). It would help to contain internals (we even use "knowledge" about key format and prefixes in here) to the place that manages them. Otherwise it will be extremely hard to maintain the code...